# PR

Pada tugas sesi ini kalian akan mengerjakan beberapa soal terkait dengan statistika deskriptif dan inferensi statistik.

Data yang digunakan adalah data hasil ujian murid sebelum vs setelah mendapatkan pelatihan modul khusus.

Fokus dari analisis adalah ingin mengetahui efektivitas dari pemberian modul tambahan.

In [2]:
import numpy as np
import pandas as pd

In [3]:
data = """Student,Pre-module,Post-module
1,18,22
2,21,25
3,16,17
4,22,24
5,19,16
6,24,29
7,17,20
8,21,23
9,23,19
10,18,20
11,14,15
12,16,15
13,16,18
14,19,26
15,18,18
16,20,24
17,12,18
18,22,25
19,15,19
20,17,16"""

from io import StringIO
df = pd.read_csv(StringIO(data))
df

Student  Pre-module  Post-module
0         1          18           22
1         2          21           25
2         3          16           17
3         4          22           24
4         5          19           16
5         6          24           29
6         7          17           20
7         8          21           23
8         9          23           19
9        10          18           20
10       11          14           15
11       12          16           15
12       13          16           18
13       14          19           26
14       15          18           18
15       16          20           24
16       17          12           18
17       18          22           25
18       19          15           19
19       20          17           16

# Soal 1
Hitung statistik deskriptif dari nilai ujian setelah vs sebelum mengikuti modul.
Apa saja yang bisa disimpulkan?

In [4]:
print(df[['Pre-module', 'Post-module']].describe())

       Pre-module  Post-module
count   20.000000    20.000000
mean    18.400000    20.450000
std      3.152276     4.058454
min     12.000000    15.000000
25%     16.000000    17.750000
50%     18.000000    19.500000
75%     21.000000    24.000000
max     24.000000    29.000000


Dari output tersebut, dapat disimpulkan bahwa:

*   Rata-rata nilai ujian setelah mengikuti modul (20.45) lebih tinggi dibandingkan dengan rata-rata nilai ujian sebelum mengikuti modul (18.40).
*   Standar deviasi nilai ujian setelah mengikuti modul (4.06) juga lebih tinggi dibandingkan dengan standar deviasi nilai ujian sebelum mengikuti modul (3.15).
*   Nilai minimum dan maksimum untuk kedua kolom juga berbeda, dengan nilai minimum dan maksimum untuk nilai ujian setelah mengikuti modul lebih tinggi dibandingkan dengan nilai ujian sebelum mengikuti modul.
*   Kuartil juga memberikan informasi tambahan tentang distribusi data. Kita dapat melihat bahwa 25% dari siswa memiliki nilai sebelum mengikuti modul di bawah 16, sedangkan 25% siswa lainnya memiliki nilai di atas 21. Sementara itu, 25% siswa memiliki nilai setelah mengikuti modul di bawah 17.75, dan 25% lainnya memiliki nilai di atas 24.

Dengan demikian, dari statistik deskriptif tersebut, kita dapat menyimpulkan bahwa pengaruh modul terhadap peningkatan nilai siswa terlihat signifikan, namun masih ada variasi dalam peningkatan tersebut.

# Soal 2

Paired T-test adalah varian dari T-test dimana sebuah sampel akan diukur sebanyak 2 kali (biasanya sebelum vs setelah suatu *treatment*).

Lakukan uji hipotesis apakah nilai ujian murid menjadi lebih baik setelah pemberian modul atau tidak.

Gunakan code dari dokumentasi berikut
https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.ttest_rel.html

In [5]:
# A> Tulis H0 dan H1
# H0 : Nilai ujian murid tidak menjadi lebih tinggi setelah pemberian modul
# H1 : Nilai ujian murid menjadi lebih tinggi setelah pemberian modul

# B> Tulis kode untuk melakukan test
from scipy.stats import ttest_rel

# Menghitung nilai T dan p-value
T, p_value = ttest_rel(df['Pre-module'], df['Post-module'])

# Menampilkan hasil uji hipotesis
alpha = 0.05
print(f"T-score: {T:.2f}")
print(f"p-value: {p_value:.5f}")
if p_value < alpha:
    print("Ada bukti menolak H0. Nilai ujian murid menjadi lebih tinggi setelah pemberian modul.")
else:
    print("Tidak ada bukti untuk menolak H0. Nilai ujian murid tidak menjadi lebih tinggi setelah pemberian modul.")

T-score: -3.23
p-value: 0.00439
Ada bukti menolak H0. Nilai ujian murid menjadi lebih tinggi setelah pemberian modul.


#C> TULIS KESIMPULAN ANDA
Nilai p-value (0.00439) < alpha (0.05), maka kita dapat menolak hipotesis nol (H0) dan menyimpulkan bahwa terdapat bukti bahwa nilai ujian murid menjadi lebih baik setelah pemberian modul.

# THANK YOU